### PR TITLE
Added `Mutex::isAcquired()` to check if lock is currently acquired

### PR DESCRIPTION
**Pull request by rhertogh**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | -

### CODE REVIEW

**Comment by rhertogh** [comment]
The purpose is to support checking if a mutex is acquired. We need a `isMutexAcquired($name)` function which would return `true` or `false`. I could also include it in this pull request but since Yii2 doesn't accept new features I left it out of the PR.
If the `isMutexAcquired($name)` function would be accepted in the PR I would be happy to include it with some test coverage.

**Comment by samdark** [comment]
What's the use case for checking if mutex is acquired without acquiring it?

**Comment by rhertogh** [comment]
It would be used in a model's save function to ensure it is only called when the mutex is active. The mutex itself would be acquired in different places and wraps different models.

**Comment by samdark** [comment]
I'd better add a method for this case. Better than exposing a property.

**Comment by rhertogh** [comment]
ok, I'll implement the method.

File: framework/mutex/Mutex.php [hunk 1]
```diff
@@ -97,6 +97,16 @@ public function release($name)
         return false;
     }
 
+    /**
+     * Checks if a lock is currently acquired
+     *
+     * @param string $name of the lock to check
+     * @return bool Returns true if currently acquired
```
Comment (samdark) [inline framework/mutex/Mutex.php:104]: Need `@since 2.0.36`

**Review by samdark — APPROVED** [review]

File: framework/mutex/Mutex.php (hunk 1 above)
Comment (rhertogh) [reply framework/mutex/Mutex.php:104]: fixed

**Comment by samdark** [comment]
Thanks!

**Pull request merged**